proof_range.go

```diff
@@ -231,7 +235,7 @@ func (t *Tree) getRangeWithProof(keyStart, keyEnd []byte, limit int) (
 	}
 	t.root.hashWithCount() // Ensure that all hashes are calculated.
 
-	rangeProof = &KeyRangeProof{RootHash: t.root.hash}
+	rangeProof = &KeyRangeProof{RootHash: t.root.hash, Version: t.root.version}
```

Comment:
but why do we need the version on the root?

Isn't enough just to include version in the leaf to differentiate them, then any inner/root node automatically changes if a key updates, even to the same value

Reply:
Now that I look at this, I think there's a flaw in how we do range proofs with versioned keys. There's missing information: we don't have the version of the leaf nodes we're trying to check, so if they are all different versions, the proof won't verify. It only verifies here because the keys happen to have the same version as the root..

Reply:
Yeah, we need to look into proofs and versions deeper. But let's make this a different pr. Fo now, it is fine.

Reply:
Sounds good.  So does that mean range proofs are temporarily broken?

Reply:
Yeah, or rather, they never worked properly with historical queries.